cmd/notation/internal/flag/options.go

```diff
@@ -123,3 +142,38 @@ func (opts *SecureFlagOpts) Credential() auth.Credential {
 		Password: opts.Password,
 	}
 }
+
+// OutputFormatFlagOpts cmd opts for output format.
+type OutputFormatFlagOpts struct {
+	// CurrentFormat is the current output format in type of string.
+	// Its default value is set by ApplyFlags.
+	CurrentFormat string
+
+	// allowedFormats is the allowed output formats set by ApplyFlags.
+	allowedFormats []output.Format
+}
+
+// ApplyFlags sets up the options for the output format flag.
+//
+// The defaultType is the default format type.
+// The otherTypes are additional format types that are allowed.
+func (opts *OutputFormatFlagOpts) ApplyFlags(fs *pflag.FlagSet, defaultType output.Format, otherTypes ...output.Format) {
+	opts.CurrentFormat = string(defaultType)
+	opts.allowedFormats = append(otherTypes, defaultType)
+
+	var quotedAllowedTypes []string
+	for _, t := range opts.allowedFormats {
+		quotedAllowedTypes = append(quotedAllowedTypes, fmt.Sprintf("'%s'", t))
+	}
+	usage := fmt.Sprintf("output format, options: %s", strings.Join(quotedAllowedTypes, ", "))
+	// apply flags
+	fs.StringVarP(&opts.CurrentFormat, "output", "o", opts.CurrentFormat, usage)
```

Comment:
Then this line will be
```suggestion
	fs.StringVarP((*string)(&opts.CurrentFormat), "output", "o", string(opts.CurrentFormat), usage)
```

Reply:
Good idea! Let me try it.